komposable-architecture/src/main/java/com/toggl/komposable/internal/CompositeReducer.kt

```diff
@@ -1,14 +1,15 @@
 package com.toggl.komposable.internal
 
+import com.toggl.komposable.architecture.NoEffect
 import com.toggl.komposable.architecture.ReduceResult
 import com.toggl.komposable.architecture.Reducer
-import com.toggl.komposable.extensions.noEffect
+import com.toggl.komposable.extensions.mergeWith
 
 internal class CompositeReducer<State, Action>(private val reducers: List<Reducer<State, Action>>) :
     Reducer<State, Action> {
     override fun reduce(state: State, action: Action): ReduceResult<State, Action> =
-        reducers.fold(ReduceResult(state, noEffect())) { accResult, reducer ->
+        reducers.fold(ReduceResult(state, NoEffect)) { accResult, reducer ->
             val result = reducer.reduce(accResult.state, action)
-            ReduceResult(result.state, accResult.effects + result.effects)
+            ReduceResult(result.state, accResult.effect mergeWith result.effect)
```

Comment:
You didn't do operator overloading here, which further enhances my point about not doing it for the effect interface

Reply:
Yeah, I feel like `+` means something slightly different than merge does.